core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -106,26 +104,41 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
   // Authentication helper used when serving method calls via socket from Python side.
   private lazy val authHelper = new SocketAuthHelper(conf)
 
+  // each python worker gets an equal part of the allocation. the worker pool will grow to the
+  // number of concurrent tasks, which is determined by the number of cores in this executor.
+  private def getWorkerMemoryMb(mem: Option[Long], cores: Int): Option[Long] = {
+    mem.map(_ / cores)
```

Comment:
Can I ask why we divide Python worker's memory by the number of cores?

Reply:
this was preexisting functionality (https://github.com/apache/spark/pull/28085/files#diff-6bc32eb2bef385137d7c16fc2c75e8b4L88) , I just changed to make it work with the resource profiles. From my understanding its just splitting the memory equally because you get a python worker per task. I thought the comment did decent job of relaying that, but maybe we need to clarify?    Thinking about this some more, there might actually be a bug here (was here before my changes) if the spark.task.cpus is > 1 because the max tasks you get couldn't be equal to number of cores, so your are splitting the memory to much.  I can file a separate jira to look at that though.

Reply:
Yes ... let's just loop this in a separate JIRA if you don't mind.

I wary of this Python memory configuration which needed a bunch of followups and separate investigation (see SPARK-25004) when it was first added. This configuration also made Spark 2.4.0 useless on Windows (SPARK-26080).

This configuration is even incomplete and conflicts with 'spark.python.worker.memory' configuration, see also SPARK-26679.

Reply:
https://issues.apache.org/jira/browse/SPARK-31444